app/components/contacts/AddNewContact.js

```diff
@@ -0,0 +1,222 @@
+    errorFieldName: 'emailErrorMsg'
+  }
+];
+
+// $FlowStyledIssue
+const Wrapper = styled.div`
+  width: 100%;
+  ${({ showBorder }) =>
+    showBorder &&
+    `
+    border: 1px solid ${smColors.borderGray}
+  `}
+  padding: 36px 22px;
+`;
+
+const FieldWrapper = styled.div`
+  margin-bottom: 28px;
+`;
+
+const TextWrapper = styled.div`
+  margin-bottom: 16px;
+`;
+
+const BoldText = styled.span`
+  font-size: 14px;
+  font-weight: bold;
+  line-height: 19px;
+  color: ${smColors.lighterBlack};
+`;
+
+const TitleWrapper = styled.div`
+  margin-bottom: 32px;
+`;
+
+const Title = styled(BoldText)`
+  color: ${smColors.darkGray};
+`;
+
+// $FlowStyledIssue
+const ButtonWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: ${({ alignment }) => (alignment === 'right' ? 'flex-end' : 'flex-right')};
+`;
+
+type Field = {
+  title: string,
+  placeholder: string,
+  fieldName: string,
+  errorFieldName: string
+};
+
+type Props = {
+  modalMode?: boolean,
+  addToContacts: Action,
+  publicWalletAddress?: string,
+  resolve: ({ publicWalletAddress: string, nickname: string, email?: string }) => void
+};
+
+type State = {
+  publicWalletAddress: string,
+  nickname: string,
+  email?: string,
+  addressErrorMsg: ?string,
+  nicknameErrorMsg: ?string,
+  emailErrorMsg: ?string,
+  renderKey: number
+};
+
+class AddNewContact extends Component<Props, State> {
+  state = { ...initialState };
+
+  static getDerivedStateFromProps(props, prevState) {
+    if (props.publicWalletAddress && !prevState.publicWalletAddress) {
+      return { publicWalletAddress: props.publicWalletAddress };
+    }
+    return null;
+  }
+
+  render() {
+    const { modalMode } = this.props;
+    const { publicWalletAddress, nickname, addressErrorMsg, nicknameErrorMsg, emailErrorMsg, renderKey } = this.state;
+    const isSaveButtonDisabled = !publicWalletAddress || !nickname || !!addressErrorMsg || !!nicknameErrorMsg || !!emailErrorMsg;
+    return (
+      <Wrapper showBorder={!modalMode} key={`render_key_${renderKey}`}>
+        {!modalMode && (
+          <TitleWrapper>
+            <Title>Add New Contact</Title>
+          </TitleWrapper>
+        )}
+        {this.renderFields()}
+        <ButtonWrapper alignment={modalMode ? 'right' : 'left'}>
+          <SmButton text="Save to contacts" isDisabled={isSaveButtonDisabled} theme="orange" onPress={this.handleSave} style={{ width: 164 }} />
+        </ButtonWrapper>
+      </Wrapper>
+    );
+  }
+
+  renderFields = () => {
+    return <React.Fragment>{fields.map((field: Field) => this.renderSingleField(field))}</React.Fragment>;
+  };
+
+  renderSingleField = ({ title, placeholder, fieldName, errorFieldName }: Field) => {
+    const { publicWalletAddress } = this.props;
+    const { addressErrorMsg, nicknameErrorMsg, emailErrorMsg } = this.state;
+    const isPublicAddressFromProps = fieldName === 'publicWalletAddress' && !!publicWalletAddress;
+    const errorMessages = {
```

Comment:
errorFieldName = 'addressErrorMsg' | 'nicknameErrorMsg' | 'emailErrorMsg'
why you create this object errorMessages? Instead use this.state[errorFieldName]

Reply:
OK